src/modules/utils/motordrivercontrol/MotorDriverControl.cpp

```diff
@@ -189,7 +235,15 @@ bool MotorDriverControl::config_module(uint16_t cs)
         this->register_for_event(ON_SECOND_TICK);
     }
 
-    THEKERNEL->streams->printf("MotorDriverControl INFO: configured motor %c (%d): as %s, cs: %04X\n", axis, id, chip==TMC2660?"TMC2660":chip==DRV8711?"DRV8711":"UNKNOWN", (spi_cs_pin.port_number<<8)|spi_cs_pin.pin);
+    //finish driver setup
+    if(chip == TMC2208) {
+        THEKERNEL->streams->printf("MotorDriverControl INFO: configured motor %c (%d): as %s, tx: %04X, rx: %04X\n", axis, id, "TMC2208", (sw_uart_tx_pin->port_number<<8)|sw_uart_tx_pin->pin, (sw_uart_rx_pin->port_number<<8)|sw_uart_rx_pin->pin);
+        //we don't want to use soft UART on a regular basis, so we delete unused UART object
+        setup = false;
+        delete this->serial;
```

Comment:
You cannot do this as the serial will be needed for the periodic alarm checks etc. so it cannot be deleted.

Reply:
ok, however the bool is maintained to not send/receive any byte via UART, which is our point since the beginning

Reply:
No You cannot stop serial being used during operation, if you do then this whole thing is useless and the pins should be used to configure this chip instead of the serial, and this PR is not needed.

Reply:
so i will keep uart being fully operating the whole time like spi or with a few restrictions? In fact, I just added full support to this driver as it's in other spi drivers but if serial is not used after setup, most of the support would be useless